packages/engine/Source/Scene/Model/ImageBasedLightingPipelineStage.js

```diff
@@ -47,7 +62,18 @@ ImageBasedLightingPipelineStage.process = function (
       );
     }
 
-    if (defined(imageBasedLighting.sphericalHarmonicCoefficients)) {
+    if (defined(specularEnvironmentMapAtlas)) {
+      shaderBuilder.addDefine(
+        "COMPUTE_POSITION_WC_ATMOSPHERE",
+        undefined,
+        ShaderDestination.BOTH,
+      );
+    }
+
+    if (
+      defined(sphericalHarmonicCoefficients) &&
+      defined(sphericalHarmonicCoefficients[0])
```

Comment:
It's a slightly weird syntax so completely optional but [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) does work on array access
```suggestion
      defined(sphericalHarmonicCoefficients?.[0])
```

Reply:
I have no idea why, but the `.[0]` notation makes me irrationally angry 🤣